package.json editor

arieljake opened the pull request:
Hi, I would like to offer a contribution to Nide: an editor for package.json files.

It was built in the nature of the other editors in the app.

I do have a lingering question about sync'ing files between editors.

arieljake committed:
package.json editor

arieljake committed:
debug form builder - null values cases

arieljake committed:
debug css

arieljake committed:
styling

arieljake committed:
moved preferences to separate class/html file for easy adding in the future

arieljake committed:
added engines section to package.json editor

coreh commented:
Hey @arieljake, is there a reason why this was closed? do you want me to wait for something before merging it?

coreh commented:
BTW, thanks for the patch :-D

arieljake commented:
no reason other than that I wanted to stop new commits from being added. This is my first pull request so im trying to figure out how to limit the commits i put into it.

arieljake committed:
preferences update - open/close style, auto save, notify code editor, added show-line-numbers and indent-unit

preferences dialog auto saves :input fields as cookies by the field
'name'
preferences dialog, on saving line numbers pref, updates code editors:
ui -> editor pool -> editors
preferences dialog shows/hides but does not rebuild each time anymore

NOTE: all editor types return "this" from constructor not just DIV
element anymore, and to preserve backwards-compatibility, sets the root
DIV as 'this.element'

arieljake commented:
to help with handling pull requests, should we prioritize creating some tests?
have you used zombie before?

arieljake commented:
also, if you do accept the additions, can you review my CSS styling and adjust as necessary to keep Nide beautiful! :)

arieljake committed:
attribution to http://package.json.jit.su/ for cheatsheet

also, debug engines cheat

arieljake committed:
new dialog and popup menu base classes

gear menu and code runner using new base classes
beginning of code autocompletion
